packages/shared/src/hooks/useBookmarkPost.ts

```diff
@@ -64,9 +68,26 @@ export default function useBookmarkPost<
     {
       onMutate: onRemoveBookmarkMutate,
       onError: (err, _, rollback) => rollback?.(),
-      onSuccess: () =>
-        onRemoveBookmarkTrackObject &&
-        trackEvent(onRemoveBookmarkTrackObject()),
+      onSuccess: (_, { id }) => {
+        if (onRemoveBookmarkTrackObject)
+          trackEvent(onRemoveBookmarkTrackObject());
+
+        const key = generateQueryKey(RequestKey.Bookmarks, user);
+        client.setQueryData<InfiniteData<FeedData>>(key, (feed) => {
+          if (!feed) return null;
+
+          return {
```

Comment:
Maybe you can move this logic to a utility function, since it looks like a useful thing to have 
```ts

removeNode(pages: Page[], id: string): {
   return feed?.pages?.map((edge) => ({
              ...edge,
              page: {
                ...edge.page,
                edges: edge.page.edges.filter(({ node }) => node.id !== id),
              },
            })),
}

```

Reply:
Oh, nice. That is true. Let me create one for update and filter like this.